### ISSUE TITLE
Additional checks for LSM Support

### ISSUE THREAD

**Issue by daemon1024** [kubearmor/KubeArmor#928]
## Feature Request

We currently rely on `/sys/kernel/security/lsm` to check for LSMs. This is fine on actual Hosts. But containerized hosts may not have this file and yet they could support stuff. So we should check for other ways to get the supported LSMs.

**Describe the solution you'd like**

AppArmor:
- As mentioned in https://kubernetes.io/docs/tutorials/security/apparmor/
  we can check for 0/1 in `/sys/module/apparmor/parameters/enabled`
  
BPF LSM:
- Use https://pkg.go.dev/github.com/cilium/ebpf/features to check if we can load LSM programs

Check for other possibilities as well.

This could potentially help solve our issues with Kubernetes in Docker situation since they don't have the file present.

**Comment by daemon1024** [comment]
@Chinwendu20 Would appreciate your help on this 🙌🏽  Thank You.

**Comment by Chinwendu20** [comment]
We could check for kernel config on the host. I found various options for doing that [here](https://superuser.com/questions/287371/obtain-kernel-config-from-currently-running-linux-system). After going through that, I suggest we use this:

` cat /proc/config.gz | gunzip | grep CONFIG_BPF_LSM`

To check if the host supports bpf_lsm. I tried this on minikube vm with VirtualBox driver and it gives this result:

`CONFIG_BPF_LSM=y`
